x-pack/plugins/maps/server/maps_telemetry/collectors/register.ts

```diff
@@ -31,6 +31,146 @@ export function registerMapsUsageCollector(
       geoShapeAggLayersCount: { type: 'long' },
       mapsTotalCount: { type: 'long' },
       timeCaptured: { type: 'date' },
+      layerTypes: {
+        ems_basemap: {
+          min: { type: 'long' },
+          max: { type: 'long' },
+          avg: { type: 'float' },
+          total: { type: 'long' },
```

Comment:
The objects describing the telemetry we're collecting are all the same. Should we pull these out to a const and reuse? i.e. something like-
```
const standardCollectionDescriptor = {
  min: { type: 'long' },
  max: { type: 'long' },
  avg: { type: 'float' },
  total: { type: 'long' },
};
```

Reply:
Looks like we could also pull the values from `TELEMETRY_LAYER_TYPE` for the `layerType` keys which would be nice for tracking layer types in one place. I know pulling values from TS enums isn't as clean more standard JS objects so your call.

Reply:
One of the earlier commits used the approach you suggest and I had to revert. the static telemetry-schema generation and validation tool doesn't work well with reference values, so the json needs to be hardcoded.